client/src/app/core/ui-services/personal-note.service.ts

```diff
@@ -66,4 +66,38 @@ export class PersonalNoteService {
             await this.http.put(`rest/users/personal-note/${pnObject.id}/`, pnObject);
         }
     }
+
+    /**
+     * Sets the 'favorite' status for several models of a type in bulk
+     *
+     * @param models The model the content belongs to
+     * @param star The new 'favorite' status
+     */
+    public async bulkSetStar(models: BaseModel[] | BaseViewModel[], star: boolean): Promise<void> {
+        if (!models.length) {
+            throw new Error('No models selected');
```

Comment:
do a `return` instead of an error.

Reply:
Which does not make the error handling more consistent. A function which should return a Promise just returns. Smells bad imo, but I'll do it.

Reply:
Is it an error, if I provide an empty array and the method does nothing? I would say it is correct: For every item provided, the star is updated. Also: Because this is an async method, `return` wil limplicitly return a promise.